sBTC SIP

```diff
@@ -0,0 +1,166 @@
+
+## Specification
+
+### Deposits (Minting sBTC)
+An API call referencing a Bitcoin transaction initiates the deposit process that can be completed within a single Bitcoin block. The API is separate from the sBTC Bootstrap Signers but will need access to the sBTC Bootstrap Signers to let them know which transactions to pick up.
+
+**API:** The API relays deposit information to the sBTC Bootstrap Signers. The API reduces the cost for deposit on rejection, reduces the scope of sBTC, and makes the sBTC system easier to service.
+
+**Bitcoin Deposit Information:** The deposit will need to pay to taproot and be spendable by a consensus threshold of the signers. The API will need to know all of the taproot spend paths and the spend paths will need to fit a specific format that cannot be clawed back in the short term in order for the Signers to consider approving it.
+
+#### Deposit Flow
+The main steps of the sBTC Deposit flow are as follows:
+1. **Deposit request:** A bitcoin holder creates a transaction on Bitcoin.
+    - The deposit transaction contains a UTXO (deposit UTXO) spendable by sBTC Signers, with an OP_DROP payload.
+    - The payload contains the recipient address of the sBTC among other relevant info for the deposit.
+    - The relevant info could contain fee suggestion or max_fee.
+2. **Proof of deposit:** The bitcoin holder submits a proof of deposit on Stacks by invoking the Signer API.
+3. **Deposit accept:**
+4. **Deposit redeem:** The sBTC Signers redeem the deposit by consuming the deposit UTXO, consolidating it into the sBTC UTXO.
+5. **Mint:** The sBTC Signers finalize the deposit acceptance by calling the mint function in .sbtc to mint the sBTC.
+
+### Withdrawals (Redeeming sBTC)
+A clarity call initiates the withdrawal process that successfully completed after 6 sortitions paying to the specified Bitcoin address. Failure cases can be resolved earlier than 6 sortitions, but some will only fail after the sBTC Bootstrap Signer attempts to create the withdrawal transaction.
+
+**Six Sortitions for Withdrawal:** The sBTC Bootstrap signers will only trigger the withdrawal on Bitcoin after the Stacks withdrawal has been confirmed by 6 sortitions. Bitcoin transactions cannot be constructed such that they’re valid for only a specific Bitcoin fork, nor can they be forcefully taken back after they’re created, and a Bitcoin fork can cause a Stacks reorg such that the initial withdrawal request becomes invalid. The sBTC Bootstrap Signers will need to wait for the withdrawals to be sufficiently final in order to make the withdrawal on Bitcoin without risk of allowing a double withdrawal from the Stacks Blockchain. The withdrawal procedure will need to track the number of valid sortitions after the withdrawal to verify that the Stacks transaction is final, and 6 is the standard choice for Bitcoin finality.
```

Comment:
When you say "sortitions," do you mean Stacks tenures?  A sortition happens for each Bitcoin block, even if it does not start a new tenure.

Reply:
Also, why are Stacks-side confirmations required here?  We're about to switch over to Nakamoto, where Stacks forks simply don't happen.

Reply:
It's not the risk of stacks forking, it's the risk of Bitcoin forking. If there's a Bitcoin fork and the Stacks chain was correctly following it and someone made a withdrawal you can't take the Bitcoin withdrawal transaction back if the new version of the Stacks chain ultimately invalidates the withdrawal on the Stacks side.

More details here: https://github.com/stacks-network/sbtc/discussions/12

Reply:
Thanks for linking that discussion @AshtonStephens -- that makes a lot of sense.

In Nakamoto, the Stacks fork is as final as the underlying Bitcoin fork after two `BlockFound` tenure changes (i.e. two accepted block-commits).  This is because each block-commit contains the hash of the first block in the preceding tenure; the block-commit which begins tenure N+2 will anchor all of the Stacks blocks up through tenure N to the Bitcoin chain.  So, perhaps the criterion here should be that an sBTC withdrawal's tenure must be confirmed by at least two more tenures?  You can measure this with the `tenure-height` built-in global in Clarity 3.